paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PaymentSheetScreen.kt

```diff
@@ -59,8 +85,28 @@ internal fun PaymentSheetScreen(
                 PaymentSheetScreenContent(viewModel, type)
             }
         },
-        modifier = modifier,
+        modifier = modifier.onGloballyPositioned {
+            contentHeight = with(density) { it.size.height.toDp() }
+        },
     )
+
+    AnimatedVisibility(
+        visible = walletsProcessingState != null &&
+            walletsProcessingState !is WalletsProcessingState.Idle &&
+            contentVisible,
+        enter = fadeIn(),
+        exit = fadeOut(),
+    ) {
+        Box(
+            contentAlignment = Alignment.Center,
+            modifier = Modifier
+                .requiredHeight(contentHeight)
+                .fillMaxWidth()
+                .background(MaterialTheme.colors.surface.copy(alpha = 0.9f)),
+        ) {
+            ProgressOverlay(walletsProcessingState)
+        }
```

Comment:
Ideally I would test if the `ProgressOverlay` composable was being displayed but `AnimatedVisibility` does not seem to play well with `assertIsDisplayed`. I could add a custom semantic for checking this (`ProgressOverlayVisible`) and test that,